src/finalfusion/embeddings.py

```diff
@@ -426,6 +541,20 @@ def __iter__(self):
             return zip(self._vocab.words, self._storage, self._norms)
         return zip(self._vocab.words, self._storage)
 
+    def _similarity(self, query: np.ndarray, k: int,
+                    skips: Set[str]) -> List['SimilarityResult']:
+        sims = self.storage[:len(self.vocab)].dot(query)
+        skip_indices = [
+            skip for skip in (self.vocab.word_index.get(skip)
+                              for skip in skips) if skip is not None
+        ]
+        indices = sims.argsort()[::-1]
+        mask = np.isin(indices, skip_indices, invert=True, assume_unique=True)
```

Comment:
What is the cost of this?

Reply:
Looks like a bunch of sorting and array allocations, I was expecting some equivalent to 
```Py
[idx for idx in indices if idx not in skip_indices]
```